### ISSUE TITLE
Incompatible with non-QWERTY keyboards on macOS

### ISSUE THREAD

**Issue by brechtm** [pete-gordon/hivelytracker#13]
My AZERTY keyboard doesn't map nicely to a piano keyboard in Hively Tracker. With the first octave selected (F1), pressing `w` produces `D-2` instead of `C-1`. `z` produces `C-1` when it should be `C-2`.

Setting the macOS keyboard layout preference to "U.S. International - PC", things work as they should.

IIRC SDL offers an option to interpret the raw keycodes instead of the mapped ones. Perhaps this can be used to make Hively work across all keyboard layouts?

Running the Windows build using Wine suffers from the same problem. Running the Windows build on Windows using VirtualBox does **not** (but sound is distorted when Hively is running).

**Comment by brechtm** [comment]
Of course, changing the keyboard layout to US makes it more difficult to input the hexadecimal numbers in effects columns. So ideally, Hively should only use raw keycodes only when "parsing" piano keys.

**Comment by mywave82** [comment]
SDL keyboard handling in general is not ideal for handling "text". To my knowledge, the reason for API to be designed this way, is that it works on all platforms.

Different OS's have very different API's for keyboard, information about mapping etc.

How to solve this: I will suggest to make an option in the UI to select if the keyboard is querty, azerty or dvorak....
